python/pyarrow/_parquet_encryption.pxd

```diff
@@ -131,3 +131,22 @@ cdef extern from "arrow/python/parquet_encryption.h" \
             SafeGetFileDecryptionProperties(
             const CKmsConnectionConfig& kms_connection_config,
             const CDecryptionConfiguration& decryption_config)
+
+cdef extern from "arrow/dataset/parquet_encryption_config.h" namespace "arrow::dataset" nogil:
+    cdef cppclass CParquetEncryptionConfig "arrow::dataset::ParquetEncryptionConfig":
+        CParquetEncryptionConfig() except +
```

Comment:
You don't need to declare a default constructor, I think.
Also, I don't think the constructor would raise any C++ exception, so the `except +` seems pointless.

Reply:
It seems this is true!

> Note that, like C++, if the class has only one constructor and it is a nullary one, it’s not necessary to declare it.